src/gauging.jl

```diff
@@ -70,161 +118,60 @@ function vidal_gauge(
-    root_S = sqrt_diag(bond_tensors[e])
-    setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vsrc]), vsrc)
-    setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vdst]), vdst)
-
-    ψsymm_mts[pe], ψsymm_mts[reverse(pe)] = copy(ITensor[dense(bond_tensors[e])]),
-    copy(ITensor[dense(bond_tensors[e])])
+  if isnothing(cache!)
+    cache! = Ref(default_norm_cache(ψ))
   end
-
-  ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
-
-  return ψsymm, pψψsymm, ψsymm_mts
+  cache![] = update(cache![]; cache_update_kwargs...)
+  return vidalitensornetwork_preserve_cache(ψ; bp_cache=cache![], kwargs...)
 end
 
-"""Put an ITensorNetwork into the symmetric gauge and also return the message tensors (which are the diagonal bond matrices from the Vidal Gauge)"""
-function symmetric_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  svd_kwargs...,
-)
-  ψ_vidal, bond_tensors = vidal_gauge(
-    ψ;
-    eigen_message_tensor_cutoff,
-    regularization,
-    niters,
-    target_canonicalness,
-    svd_kwargs...,
-  )
-
-  return vidal_to_symmetric_gauge(ψ_vidal, bond_tensors)
+function update(ψ::VidalITensorNetwork; kwargs...)
+  return VidalITensorNetwork(ITensorNetwork(ψ; update_gauge=false); kwargs...)
 end
 
-"""Transform from the Symmetric Gauge (message tensors) to the Vidal Gauge (bond tensors)"""
-function symmetric_to_vidal_gauge(
-  ψ::ITensorNetwork,
-  pψψ::PartitionedGraph,
-  mts;
-  regularization=10 * eps(real(scalartype(ψ))),
-)
-  bond_tensors = DataGraph{vertextype(ψ),Nothing,ITensor}(underlying_graph(ψ))
+"""Function to construct the 'isometry' of a state in the Vidal Gauge on the given edge"""
+function vidal_gauge_isometry(ψ::VidalITensorNetwork, edge)
+  vsrc, vdst = src(edge), dst(edge)
+  ψ_vsrc = copy(ψ[vsrc])
 
-  ψ_vidal = copy(ψ)
-
-  for e in edges(ψ)
-    vsrc, vdst = src(e), dst(e)
-    pe = partitionedge(pψψ, NamedEdge((vsrc, 1) => (vdst, 1)))
-    bond_tensors[e], bond_tensors[reverse(e)] = only(mts[pe]), only(mts[pe])
-    invroot_S = invsqrt_diag(map_diag(x -> x + regularization, bond_tensors[e]))
-    setindex_preserve_graph!(ψ_vidal, noprime(invroot_S * ψ_vidal[vsrc]), vsrc)
-    setindex_preserve_graph!(ψ_vidal, noprime(invroot_S * ψ_vidal[vdst]), vdst)
+  for vn in setdiff(neighbors(ψ, vsrc), [vdst])
+    ψ_vsrc = noprime(ψ_vsrc * bond_tensor(ψ, vn => vsrc))
   end
 
-  return ψ_vidal, bond_tensors
-end
-
-"""Function to measure the 'isometries' of a state in the Vidal Gauge"""
-function vidal_itn_isometries(
-  ψ::ITensorNetwork,
-  bond_tensors::DataGraph;
-  edges=vcat(NamedGraphs.edges(ψ), reverse.(NamedGraphs.edges(ψ))),
-)
-  isometries = Dict()
+  ψ_vsrcdag = dag(ψ_vsrc)
+  replaceind!(ψ_vsrcdag, commonind(ψ_vsrc, ψ[vdst]), commonind(ψ_vsrc, ψ[vdst])')
```

Comment:
```suggestion
  ψ_vsrcdag = replaceind(ψ_vsrcdag, commonind(ψ_vsrc, ψ[vdst]) => commonind(ψ_vsrc, ψ[vdst])')
```
Best to avoid in-place operations on ITensors unless there is a compelling reason.

Reply:
Okay sure, I have changed that and try to avoid in-place operations from now on in development.